hazelcast-client/src/test/java/com/hazelcast/client/ClientConnectionTest.java

```diff
@@ -98,4 +105,35 @@ public void testMemberConnectionOrder() {
         Collection<Client> connectedClients2 = server2.getClientService().getConnectedClients();
         assertEquals(connectedClients2.size(), 0);
     }
+
+    @Test
+    public void testConnectionRemovedOnce() {
+        HazelcastInstance server = Hazelcast.newHazelcastInstance();
+        HazelcastInstance client = HazelcastClient.newHazelcastClient();
+        HazelcastClientInstanceImpl clientImpl = ClientTestUtil.getHazelcastClientInstanceImpl(client);
+        ClientConnectionManager connectionManager = clientImpl.getConnectionManager();
+
+
+        final AtomicInteger count = new AtomicInteger();
+
+        connectionManager.addConnectionListener(new ConnectionListener() {
+            @Override
+            public void connectionAdded(Connection connection) {
+
+            }
+
+            @Override
+            public void connectionRemoved(Connection connection) {
+                count.incrementAndGet();
+            }
+        });
+
+        final Address serverAddress = new Address(server.getCluster().getLocalMember().getSocketAddress());
+        final Connection connectionToServer = connectionManager.getConnection(serverAddress);
```

Comment:
a point in case:
this seems more way easier to read:
```java
    @Test
    public void testConnectionRemovedOnce() {
        HazelcastInstance server = Hazelcast.newHazelcastInstance();
        HazelcastInstance client = HazelcastClient.newHazelcastClient();

        CountingConnectionListener listener = registerCountingConnectionListener(client);

        Connection connection = getConnectionToServer(client, server);
        connectionManager.destroyConnection(connectionToServer);
        connectionManager.destroyConnection(connectionToServer);

        assertNoOfConnectionsRemoved("connection removed should be called only once", listener, 1);
    }
```

I can see what is it doing in a few seconds.